src/Symfony/Bundle/WebProfilerBundle/EventListener/WebDebugToolbarListener.php

```diff
@@ -110,6 +111,14 @@ public function onKernelResponse(FilterResponseEvent $event)
         $this->injectToolbar($response, $request, $nonces);
     }
 
+    public function onKernelRequest()
+    {
```

Comment:
I suggest doing that only for master requests, to avoid adding multiple callbacks in case subrequests are used.

Reply:
Done